main/inc/lib/webservices/Rest.php

```diff
@@ -1590,4 +1591,59 @@ private function encodeParams(array $additionalParams = [])
 
         return $encoded;
     }
+
+    /**
+     * @param $params
+     *
+     * @throws Exception
+     *
+     * @return integer
+     */
+    public function createSessionFromModel($modelSessionId, $sessionName, $startDate, $endDate, array $extraFields = [])
+    {
+        if (!SessionManager::isValidId($modelSessionId)) {
+            throw new Exception(get_lang('ModelSessionDoesNotExist'));
+        }
+
+        $modelSession = SessionManager::fetch($modelSessionId);
+
+        foreach ($extraFields as $k => $v) {
```

Comment:
Before any foreach, always check that the array is an array and is non-empty:
```
if (is_array($extraFields) && !empty($extraFields)) {
    foreach ...
```
This will avoid notice messages in case the variable is empty